core/chains/aptos.go

```diff
@@ -0,0 +1,90 @@
+package chains
+
+import (
+	"context"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"io"
+	"log"
+	"math/big"
+	"net/http"
+	"sort"
+	"strconv"
+	"time"
+
+	"github.com/xenowits/nakamoto-coefficient-calculator/core/utils"
+)
+
+type AptosResponse struct {
+	Data struct {
+		ActiveValidators []struct {
+			VotingPower string `json:"voting_power"`
+		} `json:"active_validators"`
+		TotalVotingPower string `json:"total_voting_power"`
+	} `json:"data"`
+}
+
+func Aptos() (int, error) {
+	var coefficient int = 1
+	var expectedTotalVotingPower int
+	var calculatedTotalVotingPower big.Int
+
+	var url string = "https://fullnode.mainnet.aptoslabs.com/v1/accounts/0x1/resource/0x1::stake::ValidatorSet"
+
+	ctx, cancelFunc := context.WithTimeout(context.Background(), 10*time.Second)
+	defer cancelFunc()
+
+	req, err := http.NewRequestWithContext(ctx, http.MethodGet, url, nil)
+	if err != nil {
+		log.Println(err)
+		return 0, errors.New("couldnt create get request for aptos")
+	}
+
+	resp, err := new(http.Client).Do(req)
+	if err != nil {
+		log.Println(err)
+		return 0, errors.New("get request failed for aptos")
+	}
+	defer resp.Body.Close()
+
+	body, err := io.ReadAll(resp.Body)
+	if err != nil {
+		return 0, err
+	}
+
+	var response AptosResponse
+	err = json.Unmarshal(body, &response)
+	if err != nil {
+		return 0, errors.New("couldnt unmarshal response for aptos")
+	}
+
+	expectedTotalVotingPower, err = strconv.Atoi(response.Data.TotalVotingPower)
+	if err != nil {
+		return 0, errors.New("failed to convert total voting power to int")
+	}
+
+	var votingPowers []big.Int
+
+	for _, ele := range response.Data.ActiveValidators {
+		val, _ := strconv.Atoi(ele.VotingPower)
+		votingPowers = append(votingPowers, *big.NewInt(int64(val)))
+	}
+
+	calculatedTotalVotingPower = *utils.CalculateTotalVotingPowerBigNums(votingPowers)
+	// These must be sorted desc in order to calculate the coefficient.
+	sort.Slice(votingPowers, func(i, j int) bool {
+		return votingPowers[i].Cmp(&votingPowers[j]) > 0
+	})
+	coefficient = utils.CalcNakamotoCoefficientBigNums(&calculatedTotalVotingPower, votingPowers)
+
+	if expectedTotalVotingPower != int(calculatedTotalVotingPower.Int64()) {
```

Comment:
`expectedTotalVotingPower` could be declared as an `int64` so that a downcast of `calculatedTotalVotingPower.Int64()` to `int` would be unnecessary (it's a concern because it could truncate the value although I don't know how realistic this scenario is).